auto_submit/lib/requests/check_pull_request.dart

```diff
@@ -51,25 +59,44 @@ class CheckPullRequest extends RequestHandler {
     final rawBody = json.decode(String.fromCharCodes(base64.decode(data))) as Map<String, dynamic>;
     final PullRequest pullRequest = PullRequest.fromJson(rawBody);
     log.info('Got the Pull Request ${pullRequest.number} from pubsub.');
-    await pubsub.acknowledge('auto-submit-queue-sub', receivedMessage.ackId!);
 
     final RepositorySlug slug = pullRequest.base!.repo!.slug();
     final GithubService gitHub = await config.createGithubService();
+    final int prNumber = pullRequest.number!;
+    if (await shouldSkip(slug, prNumber, gitHub)) {
+      return Response.ok('Skip the pull request $prNumber now and check it later.');
+    }
+
     final _AutoMergeQueryResult queryResult = await _parseQueryData(pullRequest, gitHub);
     if (await shouldMergePullRequest(queryResult, slug, gitHub)) {
       log.info('Merge the pull request: ${queryResult.number}');
       // TODO(Kristin): Merge this PR. https://github.com/flutter/flutter/issues/100088
-      return Response.ok('Merge the pull request ${queryResult.number} in ${slug.fullName} repository.');
+      await pubsub.acknowledge('auto-submit-queue-sub', receivedMessage.ackId!);
+      return Response.ok('Merge the pull request $prNumber in ${slug.fullName} repository.');
     } else if (queryResult.shouldRemoveLabel) {
       log.info('Removing label for commit: ${queryResult.sha}');
+      await pubsub.acknowledge('auto-submit-queue-sub', receivedMessage.ackId!);
       await _removeLabel(queryResult, gitHub, slug, config.autosubmitLabel);
       return Response.ok('Remove the autosubmit label for commit: ${queryResult.sha}.');
     } else {
-      log.info('The pull request ${queryResult.number} has unfinished tests,'
+      log.info('The pull request $prNumber has unfinished tests,'
           'push to pubsub and check later.');
-      await pubsub.publish('auto-submit-queue', pullRequest);
     }
-    return Response.ok('Does not merge the pull request ${queryResult.number}.');
+    return Response.ok('Does not merge the pull request $prNumber.');
+  }
+
+  /// Skip commits that are less than injected latency.
+  Future<bool> shouldSkip(RepositorySlug slug, int number, GithubService gitHub) async {
```

Comment:
Check pull request is starting to contain a lot of logic. We should split this logic into a set of composable rules that we pull from a yaml config first.